src/hooks/use-execute-cli.ts

```diff
@@ -0,0 +1,71 @@
+import { __, sprintf } from '@wordpress/i18n';
+import { useCallback, useState } from 'react';
+import { getIpcApi } from '../lib/get-ipc-api';
+
+export function useExecuteWPCLI(
+	content: string,
+	projectPath: string | undefined,
+	updateMessage:
+		| ( (
+				id: number,
+				content: string,
+				output: string,
+				status: 'success' | 'error',
+				time: string
+		  ) => void )
+		| undefined,
+	id: number | undefined
+) {
+	const [ cliOutput, setCliOutput ] = useState< string | null >( null );
+	const [ cliStatus, setCliStatus ] = useState< 'success' | 'error' | null >( null );
+	const [ cliTime, setCliTime ] = useState< string | null >( null );
+	const [ isRunning, setIsRunning ] = useState( false );
+
+	const handleExecute = useCallback( async () => {
+		setIsRunning( true );
+		const startTime = Date.now();
+		const args = content.split( ' ' ).slice( 1 );
+		const result = await getIpcApi().executeWPCLiInline( {
+			projectPath: projectPath || '',
+			args,
+		} );
+
+		setTimeout( () => {
+			const msTime = Date.now() - startTime;
+			if ( result.stderr ) {
+				setCliOutput( result.stderr );
+				setCliStatus( 'error' );
+			} else {
+				setCliOutput( result.stdout );
+				setCliStatus( 'success' );
+			}
+			const completedIn = sprintf(
+				__( 'Completed in %s seconds' ),
+				( msTime / 1000 ).toFixed( 2 )
+			);
+			setCliTime( completedIn );
+			setIsRunning( false );
+
+			if ( updateMessage && id !== undefined ) {
+				updateMessage(
+					id,
+					content,
+					result.stdout || result.stderr,
+					result.stderr ? 'error' : 'success',
+					completedIn || ''
+				);
+			}
+		}, 2300 );
+	}, [ content, id, projectPath, updateMessage ] );
+
+	return {
```

Comment:
[Suggestion]: I find it a bit repetitive. As we call it like this:
```
	const {
			cliOutput,
			cliStatus,
			cliTime,
			isRunning,
			handleExecute,
			setCliOutput,
			setCliStatus,
			setCliTime,
		} = useExecuteWPCLI( content, projectPath, updateMessage, id );

```
it is clear enough that this comes from cli.
I suggest to rename them to:
```
output,
status,
time
setOutput,
setStatus,
setTime
```